### PR TITLE
Remove `AcceleratorConnector.root_gpu` and deprecate `Trainer.root_gpu`

### LINKED ISSUE #11994: Deprecate `Trainer.root_gpu`
## Proposed refactor

Deprecate this property: 
https://github.com/PyTorchLightning/pytorch-lightning/blob/cf64f3443474a93d23b5afb0417e4a60298006e6/pytorch_lightning/trainer/trainer.py#L1982-L1984

### Motivation

This property is redundant with the Strategy's `root_device` property, which generalizes this check. 
Following https://github.com/PyTorchLightning/pytorch-lightning/pull/11758, this property is not used anywhere else in the framework. This is only used in tests, which can be updated to use `trainer.strategy.root_device` instead

### Pitch

Deprecate this property, and recommend using `trainer.strategy.root_device` instead
Regarding how to deprecate the property, this issue provides an example: https://github.com/PyTorchLightning/pytorch-lightning/issues/10615

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @justusschock @awaelchli @rohitgr7 @tchaton @akihironitta @kaushikb11 @ananthsub @ninginthecloud @jjenniferdai

### PR DESCRIPTION
**Pull request by DuYicong515**
## What does this PR do?

Part of #11449: remove unused properties in accelerator_connector: root_gpu
Part of #12171: deprecate device related properties in Trainer in favor of getting them from `Trainer.accelerator` or `Trainer.strategy`: root_gpu

AcceleratorConnector.root_gpu was not meant to be public. It's removed from AcceleratorConnector and deprecated in Trainer in favor of using `Trainer.strategy.root_device.index if isinstance(Trainer.accelerator, GPUAccelerator) else None`

Fixes #11994

### Does your PR introduce any breaking changes? If yes, please list them.

N/A

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @four4fish

### CODE REVIEW

**Review by four4fish — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -2016,7 +2016,11 @@ def num_processes(self) -> int:
 
     @property
     def root_gpu(self) -> Optional[int]:
-        return self._accelerator_connector.root_gpu
+        rank_zero_deprecation(
+            "`Trainer.root_gpu` is deprecated in v1.6 and will be removed in v1.8. Please use "
+            "`Trainer.strategy.root_device.index if isinstance(Trainer.accelerator, GPUAccelerator) else None` instead."
```
[the remark this answers is not in the document]
Comment (DuYicong515) [reply pytorch_lightning/trainer/trainer.py]: Updated! Thanks!

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 2]
```diff
@@ -775,14 +775,6 @@ def parallel_devices(self) -> List[Union[torch.device, int]]:
     def num_processes(self) -> int:
         return self.devices if self.devices is not None else 1
 
-    @property
```
Comment (Borda) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:778]: just thinking if all connectors shall be protected classes...

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -503,6 +503,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `LightningDataModule.on_save/load_checkpoint` in favor of `state_dict/load_state_dict` ([#11893](https://github.com/PyTorchLightning/pytorch-lightning/pull/11893))
 
 
+- Deprecated `Trainer.root_gpu` in favor of `Trainer.strategy.root_device.index` when GPU is used. ([#12262](https://github.com/PyTorchLightning/pytorch-lightning/pull/11978))
```
Comment (akihironitta) [inline CHANGELOG.md:506]:
```suggestion
- Deprecated `Trainer.root_gpu` in favor of `Trainer.strategy.root_device.index` when GPU is used. ([#12262](https://github.com/PyTorchLightning/pytorch-lightning/pull/12262))
```

**Review by akihironitta — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (akihironitta) [reply CHANGELOG.md:521]: The link is pointing to an unrelated PR.

Comment (DuYicong515) [reply CHANGELOG.md]: ah sorry, good catch, fixed!

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 2 above)
Comment (ananthsub) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:791]: they already are: https://github.com/PyTorchLightning/pytorch-lightning/issues/9778
